### DISCUSSION TITLE
Right buttom menu shifted

### DISCUSSION THREAD

**Discussion by studzinski55** [kangyu-california/PersistentWindows#113] [General]
When auto-restore is enabled, the right click menus of taskbar icons are all shifted to the same right side of the taskbar.
Very frustrating...  Is any way to fix it?

**Comment by kangyu-california** [comment]
There is a problem with PW context menu location, which is fixed just now.

If you have issue with other icon menus, please attach a screen shot to illustrate the issue. Because I don't think PW would affect the location of context menu of other apps.

Thanks!

**Reply by studzinski55** [reply]
As you can see, I could save a snapshot. In this case, it concerns the "right button" Firefox icon context menu on the taskbar.
![image](https://user-images.githubusercontent.com/83953783/118395898-2d14b180-b64d-11eb-9799-c57f3ddf31ed.png)

**Comment by studzinski55** [comment]
As I could test, for instant there is the only case (taskbar icons context menu).
Cordially
André

**Reply by kangyu-california** [reply]
The shifted context menu issue only manifest when right mouse click on PW icon, which is fixed in 5.23

**Reply by studzinski55** [reply]
This issue is not fixed in 5.23 yet. :-(
Now, the right click menu is shifted always to the same place. Right or left, depending to the taskbar icon position. 
Unfortunately, while this menu is shifted horizontally, it is also shifted down, in the way that a part of menu remains inaccessible in any case. :-(  This issue concerns **all context menus** on task bar icons, **but just not PM icon**...
I cannot take a screenshot, because when I select something else, this menu disappears, of course. 
Cordially yours
Andre

**Comment by studzinski55** [comment]
As you can see, I could save a snapshot. In this case, it concerns the "right button" Firefox icon context menu on the taskbar.
![image](https://user-images.githubusercontent.com/83953783/118395898-2d14b180-b64d-11eb-9799-c57f3ddf31ed.png)

**Reply by kangyu-california** [reply]
The issue is reproducible on WIn7 but not Win10, luckily there is a workaround, run PW with -fix_unminimized_window=0

Have fun!

**Comment by studzinski55** [comment]
Unfortunately the fix in version 5.23 is not valid for Win 7. The proposed workaround does not work either... 
When snapshot is restored, the context menus on taskbar shift again...
 :-(

**Reply by kangyu-california** [reply]
Sorry I can not confirm your claim, -fix_unminimized_window=0 works perfectly on my Win7, note there is no space in the command option.

I have also found a  Win7 bug that might explain your issue.

Win7 task scheduler does not actually kill PW process when select "End" operation, this leaves old PW process alive, preventing new PW process from starting.
When this happens, all taskbar icons will shift left, exposing the dialog icon of failing new PW process

You have to kill old PW process from task manager, also click off the failing dialog of new PW process (which says can not start a second PW process)

**Comment by studzinski55** [comment]
Your app (on win7)
        worked well till few weeks ago :-)
      
    After last up-dates, it
        arranges icons in bad way again (leaves empty rows)  :-(
      
    
    
    
    On 18.05.2021 21:34:24,
      kangyu-california wrote:
    
    
      
      Just found a Win7 bug that task scheduler does not actually
        kill PW process when select "End" operation, this leaves old PW
        process alive, preventing new PW process from starting.
        When this happens, all taskbar icons will shift left, exposing
        the dialog icon of failing new PW process
      You have to kill old PW process from task manager, also click
        off the failing dialog of new PW process.
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/kangyu-california/PersistentWindows/discussions/113#discussioncomment-754678",
"url": "https://github.com/kangyu-california/PersistentWindows/discussions/113#discussioncomment-754678",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by kangyu-california** [comment]
Pls understand root cause issue on Windows7 is impractical.
If older PW release worked on Windows7, don't upgrade to newer version.
